### PR TITLE
resolve nuclear "Full screen visualizer opacity #1476"

### PR DESCRIPTION
**Pull request by HaraldFair**
Please review my pull request to resolve issue #1476.

The linter has been run.

Testing has been run.

I spend significant time looking into / writing a test to support this new functionality but it seems that since Jest runs in NodeJS it doesn't support CSS properties except under artificial (inline styling) conditions that would be inappropriate for this feature.

The existing snapshot tests complete with no errors.

### CODE REVIEW

File: packages/ui/lib/components/VisualizerOverlay/index.tsx [hunk 1]
```diff
@@ -21,32 +21,56 @@ const VisualizerOverlay: React.FC<VisualizerOverlayProps> = ({
   exitFullscreenLabel,
   isFullscreen = false
 }) => {
-  const presetOptions = presets.map(preset => ({
+  // State for managing overlay opacity
+  const [isHovered, setIsHovered] = useState(false);
+
+  // Establish timer for overlay opacity
+  useEffect(() => {
+    let timeoutId: NodeJS.Timeout;
```
Comment (nukeop) [inline packages/ui/lib/components/VisualizerOverlay/index.tsx:29]: Nuki is right that you should use the browser-side type here

File: packages/ui/lib/components/VisualizerOverlay/index.tsx [hunk 2]
```diff
@@ -21,32 +21,56 @@ const VisualizerOverlay: React.FC<VisualizerOverlayProps> = ({
...
-            basic
-            icon='expand'
-            onClick={onEnterFullscreen} 
-          />
-        )
-    }
-  </div>;
+  return (
+    <div
+      className={`${styles.visualizer_overlay}${
```
Comment (nukeop) [inline packages/ui/lib/components/VisualizerOverlay/index.tsx:51]: Use the `classnames` package to combine classes. It can also handle conditions

**Review by nukeop — CHANGES_REQUESTED** [review]
Thanks, this is a pretty good way to solve it. I only have 2 minor comments. Applying Nuki's changes is not mandatory!

File: packages/ui/lib/components/VisualizerOverlay/index.tsx (hunk 2 above)
Comment (HaraldFair) [reply packages/ui/lib/components/VisualizerOverlay/index.tsx:51]: Thank you, I will make the requested changes and update my PR.

File: package.json [hunk 3]
```diff
@@ -38,6 +38,7 @@
     "@typescript-eslint/eslint-plugin": "^4.33.0",
     "@typescript-eslint/parser": "^4.33.0",
     "babel-eslint": "^10.0.3",
+    "classnames": "^2.5.1",
```
Comment (nukeop) [inline package.json:41]: classnames is already installed in projects that need it. You don't have to install it, so it should be removed here.

File: packages/ui/lib/components/VisualizerOverlay/index.tsx [hunk 4]
```diff
@@ -21,32 +23,65 @@ const VisualizerOverlay: React.FC<VisualizerOverlayProps> = ({
...
+  const stylesMod = classNames.bind(styles);
+
+  const ovClasses = stylesMod({
+    visualizer_overlay: true,
+    hover: isHovered
+  });
+
+  return (
+    <div
+      className={ovClasses}
```
Comment (nukeop) [inline packages/ui/lib/components/VisualizerOverlay/index.tsx:64]:
```suggestion
      className={classnames(styles.visualizer_overlay, {
          [styles.hover]: isHovered
      })}
```

And then you can remove lines 55-60.

I also recommend importing `classnames` as `cx` to make it shorter.

**Review by nukeop — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (nukeop) [reply package.json:41]: Also, as there are no further changes to packages, you should remove the changes you've made to package-lock.json from the PR (but don't delete the file altogether).

**Comment by HaraldFair** [comment]
OK, I think this is compliant with the requested changes now. I've re-run tests and lint:fix and manually verified functionality after all actions.

**Pull request merged**